### ISSUE TITLE
Make the plugin thread safe

### ISSUE THREAD

**Issue by iemejia** [Zlika/reproducible-build-maven-plugin#10]
When running the plugin on a build in parallel with maven  mvn --threads 1C  I got this warning
```
[WARNING] *****************************************************************
[WARNING] * Your build is requesting parallel execution, but project      *
[WARNING] * contains the following plugin(s) that have goals not marked   *
[WARNING] * as @threadSafe to support parallel building.                  *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in Apache Beam :: Parent:
io.github.zlika:reproducible-build-maven-plugin:0.3
``
Maybe it is a good idea to ensure that the plugin is threadsafe and add the annotation if it is the case.

**Comment by Zlika** [comment]
Thanks for the report.
I reviewed the code and nothing prevents the plugin from being used in a parallel build.
Fixed by 9a80e21.

**Comment by Zlika** [comment]
reproducible-build-maven-plugin version 0.4 has been released on Maven Central. Thank you for your contribution.
